libconsensus/pbft/PBFTEngine.h

```diff
@@ -458,7 +458,8 @@ class PBFTEngine : public ConsensusEngineBase
         /// get leader failed or this prepareReq is not broadcasted from leader
         if (!leader.first || req.idx != leader.second)
         {
-            if (!m_emptyBlockViewChange)
+            /// leader failure doesn't caused by view change should output warning
+            if (!m_leaderFailed)

```

Comment:
why do you change the condition from emptyBlockViewChange to leaderFailed ??

Reply:
During viewchange procedure,  `getLeaderFailed` returns `true`, since `non-empty-block viewchange` has been warninged by `viewchangeWarning`, I think `output getLeader failed warning` is useless.

In the other case, in the condition of external network bandwidth, in the procedure of `empty-block viewchange`, PBFTEngine may receive a `prepare` request while `viewchange not completed`, in this case,   there maybe too may `getLeader failed warnings` even if everything runs ok.